tenpy/algorithms/truncation.py

```diff
@@ -307,133 +307,101 @@ def svd_theta(theta, trunc_par, qtotal_LR=[None, None], inner_labels=['vR', 'vL'
     return U, S, VH, err, renormalization
 
 
-def _qr_theta_Y0(B_L: npc.Array, B_R: npc.Array, theta: npc.Array, move_right: bool, expand: float, min_block_increase: int):
-    """Generate the initial guess `Y0` for the (left) right isometry for the QR based theta decomposition decompose_theta_qr_based().
+def _qr_theta_Y0(T_L: npc.Array, T_R: npc.Array, theta: npc.Array, move_right: bool, expand: float, min_block_increase: int):
+    """Generate the initial guess `Y0` for the (left) right isometry for the QR based theta decomposition `decompose_theta_qr_based()`.
 
     Parameters
     ----------
-    B_L : Array with legs [vL, p, vR]
-    B_R : Array with legs [vL, p, vR]
+    T_L : Array with legs [vL, p, vR]
+        If ``move_right=True`` the tensor `T_L` must be in `'B'` or `'Th'` form.
+        If ``move_right=False`` the tensor `T_L` must be in `'A'` form.
+    T_R : Array with legs [vL, p, vR]
+        If ``move_right=True`` the tensor `T_R` must be in `'B'` form.
+        If ``move_right=False`` the tensor `T_R` must be in `'A'` or `'Th'` form.
```

Comment:
This is not correct yet. It might be tricky to find a convention for the leg charge, because:
- for the TEBD (where we always have `move_right=False`), psi is in `B` form (hence `theta_old=S@B_L@B_R`), and `vL_old = T_R.get_leg('vL')=B_R.get_leg('vL')` makes sense as a general convention. `vL_old` would then be the right leg of `S`.
- However, if we have `move_right=False` for the VariationalApplyMPO then `theta=A_L@Th_R`. Consequently we would have `vL_old = T_R.get_leg('vL')=Th_R.get_leg('vL')` which is the left leg of `S`.

This is a problem if `S` changes the charge.

Reply:
- If we keep the T_L as args, should name them old_T_L, like in decompose_theta_qr_based.
- I understood from the discussion last time that we wanted to change the args of decompose_theta_qr_based, such that it no longer gets the old tensors, but rather only the info it needs, namely the old leg and the qtotal. (Should carefully document which leg we expect!) Then we can let the engine take care of getting that leg.
- If `S` changes the charge, i.e. if there is a mixer, we dont care which leg we get, the left or right leg of S.
We want to guess a leg that allows a good decomposition of the new evolved theta. We use a leg that allowed a decomposition of the old theta as a starting point and just add a few extra basis vectors (this is reasonable, since the new theta has not changed much; small time step). *Any* leg that allows a decomposition of theta will do, so it can be the left or right leg of S, it does not matter.